### ISSUE TITLE
Rails databases: {} option

### ISSUE THREAD

**Issue by senny** [DataDog/dd-trace-rb#574]
Going through the changelogs of recent versions I found that support for multiple AR databases is now supported out of the box! 🎉 

I tried to change our setup which was based on span preprocessors to use the new `databases:` option when configuring Rails. Unfortunately, I was not able to get it to work and everything was tracked in the database service specified as `database_service`.

My configuration was something like this:

```ruby
    c.use :rails,
      database_service: 'main-db',
      databases: {
        archive: { service_name: 'archive-db' },
        follower: { service_name: 'follower-db' }
      }
```

I tried to find how `databases:` is applied but I was not able to find the right spot. (I was mainly looking at [framework.rb](https://github.com/DataDog/dd-trace-rb/blob/v0.16.0/lib/ddtrace/contrib/rails/framework.rb))

Would you be able to give me a hint where I can follow-up to get this working? Thanks a lot in advance and great work on the rapid pace of new features!

**Comment by delner** [comment]
Hey @senny , glad to here you're excited about the new database support!

I think what you have above was our original implementation of multi-DB configuration, which we reworked a bit. The current way of implementing it (I believe as of 0.14.0) is documented [here](https://github.com/DataDog/dd-trace-rb/blob/master/docs/GettingStarted.md#active-record). It's just a slightly different syntax, but requires the use of `use :active_record` instead.

```ruby
Datadog.configure do |c|
  c.use :active_record, describes: :archive, service_name: 'archive-db'
  c.use :active_record, describes: :follower, service_name: 'follower-db'
end
```

The reason for this is was the introduction of multiplexing: subdivisions of unique tracer settings by key (in this case, by each database.) in the future as the integration matures, you'll be able to configure a full set of trace settings for different DBs, instead of just `service_name`, which should allow for greater customization of trace behavior. 

Let me know if this configuration works for you!

**Comment by delner** [comment]
Yeah you're absolutely right @senny the documentation is inaccurate here. I updated the code sample in ActiveRecord, but I neglected all the surrounding information and Rails itself. I'll have to fix this to be a bit more clear.

Also yes, `use :rails` will activate ActiveRecord automatically if present, and configure it with a service name/tracer as defined in Rails (e.g. `database_service`, `tracer`). Basically, the `use` command is the same thing as `patch & apply settings`, so if `use :active_record` is called twice with configuration, the first call will apply the patch, and the second will override the settings applied by the first. Because Rails has a special `after_initialize` hook in which it activates ActiveRecord, the configuration options passed from Rails should always override those settings set in `use :active_record` as defined in the initializer file.

To us, this is suboptimal and a bit clumsy. I have some plans to try to streamline this a bit, by basically turning Rails into a "composite" integration, which is composed of sub-integrations like ActiveRecord. The hope is you'll be able to configure ActiveRecord within the Rails group instead of having to configure it separately. Something like:

```ruby
Datadog.configure do |c|
  c.use :rails do |rails|
    rails.service_name = 'my-app'

    # Settings for ActiveRecord tracing
    rails.active_record do |active_record|
      active_record.service_name = 'my-database'
    end
  end
end
```

The foundations for this were already laid in #450. Still need to figure out how to introduce the `describes` option into this, but it'll likely look pretty similar.

**Comment by delner** [comment]
Documentation for this will be fixed by #578.

**Comment by delner** [comment]
The documentation change was merged and will be released with 0.17.0. Going to close this for now. If you have any other questions or comments regarding configuring databases like this, please feel free to comment or re-open.
